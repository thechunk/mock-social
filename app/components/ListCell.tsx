import * as React from 'react';
import {PureComponent} from "react";
import {GestureResponderEvent, TouchableOpacity, View} from "react-native";

export interface IListCell {
    onPress?: (event: GestureResponderEvent) => void;
}
export default class ListCell extends PureComponent<IListCell> {
    render() {
        return (
            <TouchableOpacity onPress={this.props.onPress}>
                <View>
                    {this.props.children}
                </View>
            </TouchableOpacity>
        )
    }
}