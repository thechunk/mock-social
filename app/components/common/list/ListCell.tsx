import * as React from 'react';
import {PureComponent} from 'react';
import {TouchableOpacity, View} from 'react-native';

export default class ListCell extends PureComponent<IListCell> {
    render() {
        return (
            <View style={this.props.containerStyle}>
                <TouchableOpacity style={this.props.cellStyle} onPress={this.props.onPress}>
                    {this.props.children}
                </TouchableOpacity>
            </View>
        )
    }
}