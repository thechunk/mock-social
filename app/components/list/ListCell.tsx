import * as React from 'react';
import {PureComponent} from 'react';
import {TouchableOpacity, View} from 'react-native';

export default class ListCell extends PureComponent<IListCell> {
    render() {
        return (
            <View style={this.props.style}>
                <TouchableOpacity onPress={this.props.onPress}>
                    {this.props.children}
                </TouchableOpacity>
            </View>
        )
    }
}