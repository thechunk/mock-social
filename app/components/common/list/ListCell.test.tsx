import * as React from 'react';
import {shallow} from 'enzyme';
import ListCell from './ListCell';
import {Text} from "react-native";

describe('ListCell', () => {
    it('renders correctly', () => {
        expect(shallow(<ListCell><Text>Foo</Text></ListCell>)).toMatchSnapshot();
    });
});
