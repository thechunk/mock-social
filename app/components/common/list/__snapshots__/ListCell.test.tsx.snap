// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCell renders correctly 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
  />
</View>
`;
